_docs/setup/kubernetes/mesh-expansion.md

```diff
@@ -0,0 +1,208 @@
+
+## Installation steps
+
+Setup consists of preparing the mesh for expansion and installing and configuring each VM.
+
+An example script to help with K8S setup is available in
+[install/tools/setupMeshEx.sh](https://raw.githubusercontent.com/istio/istio/master/install/tools/setupMeshEx.sh).
+
+An example script to help configure a machine is available in [install/tools/setupIstioVM.sh](https://raw.githubusercontent.com/istio/istio/master/install/tools/setupIstioVM.sh).
+You should customize it based on your provisioning tools and DNS requirements.
+
+### Preparing the K8s for expansion
+
+* Setup internal load balancers for Kube DNS, Pilot, Mixer and CA. This step is specific to
+each cluster, you may need to add annotations.
+
+```bash
+install/tools/setupMeshEx initCluster
+```
+or
+```
+kubectl apply -f install/kubernetes/meshex.yaml
+```
+
+* Generate the Istio 'cluster.env' config to be deployed in the VMs. This file contains
+the cluster IP address ranges to intercept.
+
+```bash
+install/tools/setupMeshEx generateConfigs MY_CLUSTER_NAME
+```
+
+Example generated files:
+
+   ```bash
+   cat /usr/local/istio/proxy/cluster.env
+   ```
+   ```
+   ISTIO_SERVICE_CIDR=10.23.240.0/20
+   ```
+* Generate DNS config file to be used in the VMs. This will allow apps on the VM to resolve
+cluster service names, which will be intercepted by the sidecar and forwarded.
+
+```bash
+install/tools/setupMeshEx generateConfigs MY_CLUSTER_NAME
+```
+
+Example generated files:
+
+   ```bash
+cat /etc/dnsmasq.d/kubedns
+```
+```
+   server=/svc.cluster.local/10.128.0.6
+   address=/istio-mixer/10.128.0.7
+   address=/mixer-server/10.128.0.7
+   address=/istio-pilot/10.128.0.5
+   address=/istio-ca/10.128.0.8
+  ```
+
+### Setting up the machines
+
+* Copy the config files and istio debian files to each machine joining the cluster, as
+`/etc/dnsmasq.d/kubedns` and `/var/lib/istio/envoy/cluster.env`.
+
+* Configure and verify DNS settings - this may require installing dnsmasq, adding it to
+`/etc/resolv.conf` directly or via DHCP scripts. To verify, check that the VM can resolve
+names and connect to pilot, for example:
+
+On the VM/external host:
+```bash
+dig istio-pilot.istio-system
+```
+```
+    # This should be the same address shown as "EXTERNAL-IP" in 'kubectl get svc -n istio-system istio-pilot-ilb'
+    ...
+    istio-pilot.istio-system. 0	IN	A	10.128.0.5
+    ...
+```
+```bash
+    # Check that we can resolve cluster IPs. The actual IN A will depend on cluster config.
+    dig istio-pilot.istio-system.svc.cluster.local.
+```
+```
+    ...
+    istio-pilot.istio-system.svc.cluster.local. 30 IN A 10.23.251.121
+```
+```bash
+dig istio-ingress.istio-system.svc.cluster.local.
+```
+```
+    ...
+    istio-ingress.istio-system.svc.cluster.local. 30 IN A 10.23.245.11
+```
+
+* Verify connectivity. We'll check if the VM can connect to the pilot, and it can connect to
+an endpoint.
```

Comment:
Verify connectivity by checking whether the VM can connect to the pilot and to an endpoint.

Reply:
changed to "Verify connectivity by checking whether the VM can connect to Pilot and to an endpoint." (I think we use Pilot for Istio Pilot)